src/backend/distributed/commands/index.c

```diff
@@ -180,6 +180,8 @@ PreprocessIndexStmt(Node *node, const char *createIndexCommand,
 		return NIL;
 	}
 
+	EnsureCoordinator();
```

Comment:
Why is this change needed?

Reply:
We were missing this check for index cmds, but luckily, we already had this in ExecuteDistributedDDLJob. However, as of this PR, ExecuteDistributedDDLJob doesn't have this check anymore. So effectively this is a no-op change for index cmds.